lib/perl/Genome/InstrumentData/Command/Import/GenerateBase.pm

```diff
@@ -0,0 +1,180 @@
+package Genome::InstrumentData::Command::Import::GenerateBase;
+
+use strict;
+use warnings;
+
+use Genome;
+
+require File::Basename;
+require List::MoreUtils;
+use Params::Validate qw( :types );
+use Text::CSV;
+
+class Genome::InstrumentData::Command::Import::GenerateBase {
+    is => 'Command::V2',
+    is_abstract => 1,
+    has_input => {
+        file => {
+            is => 'Text',
+            doc => 'Comma (.csv) or tab (.tsv) separated file of entity names, attributes and other meta data. Separartor is determined by file extension.',
+        },
+    },
+    has_optional_output => {
+        output_file => {
+            is => 'Text',
+            default_value => '-',
+            doc => 'Output file to put the commands to create the needed entities for import.',
+        },
+    },
+    has_optional_transient => {
+        _output => { is => 'ARRAY', default_value => [], },
+        _instdata_property_names => { is => 'ARRAY', },
+        _entity_attributes => { is => 'ARRAY', },
+        _nomenclature => { is => 'Text', },
+        _names_seen => { is => 'HASH', default_value => {}, },
+    }
+};
+
+sub help_detail {
+    return <<HELP;
+The file should be a comma or tab separated values and indicated with the appropriate extension (csv and tsv). Column headers to use to generate the create commands should start with the entity (individual, sample, library, instdata) name then a period (.) and then then attribute name (Ex: sample.name_part). Here are some required and optional columns. For more, see each entity's create command (Ex: genome sample create --h). Please see Confluence documentation for more information and a full example.
+
+Individual\n
+ Required
+  individual.name_part => Name or id from external source.
+  individual.taxon      => Species name of the taxon.
+
+Sample\n
+ Required
+  sample.name_part => Name or id from external source. If name is given, the individual and library names will be resolved from it.
+  sample.extraction_type => 'genomic dna' or 'rna'
+
+ Optional, but recommended:
+  sample.common_name    => Usually normal or tumor to indicate disease state.
+
+Library\n
+ Optional
+  library.ext           => Extension to append to the sample name. Deault is 'extlibs'.
+ 
+Instrument Data (needed for generating source-files.tsv)
+ Required
+  instdata.source_files  => Local copy of the source files to import.
+ Optional
+  instdata.run_name      => The run name or id.
+
+HELP
+}
+
+sub entity_types {
+    return (qw/ individual sample library instdata/);
+}
+
+sub _open_file_parser {
+    my $self = shift;
+
+    my $file = $self->file;
+    my ($dir, $basename, $ext) = File::Basename::fileparse($file, 'csv', 'tsv');
+    die $self->error_message("Cannot determine type for file: %s. It needs to end with .csv or .tsv.", $file) if not $ext;
+    my $parser = Text::CSV->new({
```

Comment:
I might have hasd a reason, but forgot. I guess I didn't see the SVR adding any value here. It should use Text::CSV under the hood. I can replace it if necessary.